core/variant.cpp

```diff
@@ -3065,7 +3065,7 @@ bool Variant::hash_compare(const Variant& p_variant) const {
 				return false;
 
 			for(int i = 0; i < l.size(); ++i) {
-				if(! l[0].hash_compare(r[0]))
+				if(! l[i].hash_compare(r[i]))
```

Comment:
Not sure about this! Does both indices need to be counter `i`? or just one of them? Are we comparing if the two arrays are equal (it seems to me it is)?